### ISSUE TITLE
use with blender-actions-to-json 

### ISSUE THREAD

**Issue by kevzettler** [chinedufn/webgl-wield-item-tutorial#1]
Hey @chinedufn This example repo uses: [keyframes-to-dual-quats](https://www.npmjs.com/package/keyframes-to-dual-quats)

https://github.com/chinedufn/webgl-wield-item-tutorial/blob/master/tutorial.js#L44-L45

I'm assuming `'./cowboy-model.json'` is from `collada-dae-parser` is that correct?

Would it be possible to pass the output from `blender-actions-to-json` to `keyframes-to-dual-quats` here as well?

**Comment by chinedufn** [comment]
Ok so my laptop is fried and I'm on mobile so it's hard to give you like numbers or screenshots. I'll do my best here:

Yes it's from collada-dae-parser

You can't use keyframes-to-dual-quats (sorry for inconsistencies im figuring it out as I go as you can likely tell).

collada-dae-parser pre-multiplies all of your poses with your inverse bind matrices and transposes them for you as well so that they go from two major Blender to column major OpenGL

blender-actions-to-json does neither for you. It just gives your data straight from blender.

So you could certainly iterate over all of your actions and multiply them by their inverse bind matrix, then transpose them. Then pass these new matrices into keyframes-to-dual-quats 

---

If you look at the websocket onmessage handler in this link below we're iterating over all of our actions and multiplying them by the inverse bind matrix and then transposing them. Then converting them into a dual wuaternion.

So tl;dr your quickest bet is to take the armature.actions = Object.keys(armature.actions) ... reduce function. It'll show you what I mean by multiplying by inverse bind and transposing

Do let me know how I can better clarify! 

https://github.com/chinedufn/blender-webgl-skinned-hot-reload-experiment/blob/master/viewer.js

**Comment by chinedufn** [comment]
line numbers* 

But yeah the reduce call that I mentioned from that link will convert your blender-actions into dual quats in the same way that collada-dae-parser + keyframes-to-dual-quats would have
